### ISSUE TITLE
OSX can not load DroidSans.ttf in XCode

### ISSUE THREAD

**Issue by momofarm** [erincatto/Box2D#417]
When I try to build and run TestBed in XCode, the build is OK.

But after I fix the openGL version problem (due to OSX restriction to openGL version?),
the code simply can not found the DroidSans.ttf in it's relative directory (the original path is a 
relative path :"../Data/DroidSans.ttf")

The problem lies in XCode build it's output application is _NOT_ lies in the build directory like linux, which is near the source repository. When build with XCode, the generated code is located somewhere in computer (please refer to https://developer.apple.com/library/ios/recipes/xcode_help-locations_preferences/Build/Build.html for more detail) .

The problem can be fixed by: 
a. change xcode's setting, make sure the output product fit the current relative path practice. or 
b. let xcode copy the DroidSans.ttf to generated product path. but also need to change source code.

**Comment by aglv** [comment]
Setting the Testbed scheme's Working Directory option to .../Box2D/Box2D/Build/Testbed also works

**Comment by louis-langholtz** [comment]
I've seen where finding DroidSans.ttf can be a problem in MS Visual Studio solution-based builds too. I looked into the problem in the source code. A solution I came up with was algorithmically speaking:

1. Split off the font loading code from the rest of the GUI initializing code so that font loading is a separately callable function (or method) and the rest of the GUI initialization does have to have fonts loaded to still work.
2. Setup an array of relative paths to try finding the font file at.
3. Iteratively call the font loading function until it succeeds with loading from a given path.
4. Move on to the other GUI initializing function. Note that in step 1, this would still work even if no font was loaded.

With this, in the worst case, the Testbed starts up without any text displayed in its GUI. That's better, at least for me, than aborting. It also was able to solve the problem for running the Testbed right from the Windows folder it's built in for me.
